src/main/kotlin/org/rust/ide/hints/RustParameterInfoHandler.kt

```diff
@@ -0,0 +1,159 @@
+package org.rust.ide.hints
+
+import com.intellij.codeInsight.lookup.LookupElement
+import com.intellij.lang.parameterInfo.*
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.util.parentOfType
+
+/**
+ * Provides funcions/methods afruments hint.
+ */
+class RustParameterInfoHandler : ParameterInfoHandler<PsiElement, RustArgumentsDescription> {
```

Comment:
>But if you're sure it must work with RustArgListElement, I'll have a look under the hood a bit later.

No, I am not sure, let's merge this as is. Although com.intellij.codeInsight.hint.api.impls.MethodParameterInfoHandler uses more specific type there. But it is `ParameterInfoHandlerWithTabActionSupport` (so that you can use <kbd>Tab</kbd> to navigate through parameters).